feat(spm): add multi-module Package.swift auto-generation

hanrw opened the pull request:
## Summary
Add a new root-level plugin `co.touchlab.kmmbridge.spm` that automatically generates Package.swift for multi-module KMP projects.

## New Features

- **spmDevBuildAll**: Builds all XCFrameworks locally and generates Package.swift with local paths for development
- **kmmBridgePublishAll**: Publishes all modules and generates Package.swift with URLs for distribution
- **generatePackageSwift**: Generates Package.swift from published metadata

## Changes

- Add `KmmBridgeSpmPlugin` for root-level SPM management
- Add `KmmBridgeSpmExtension` for configuration options
- Add `SpmModuleMetadata` for JSON metadata exchange between modules
- Add `writeSpmMetadata` task to each module for metadata generation
- Disable module-level `spmDevBuild` when root SPM plugin is applied
- Add comprehensive documentation in docs/SPM_MULTI_MODULE.md

## Benefits

- No manual Package.swift editing required
- No need for `useCustomPackageFile` or `perModuleVariablesBlock` flags
- Unified workflow for both local development and CI publishing
- Automatic platform version resolution (takes maximum)
- Automatic Swift tools version resolution

hanrw committed:
feat(spm): add multi-module Package.swift auto-generation

Add a new root-level plugin `co.touchlab.kmmbridge.spm` that automatically
generates Package.swift for multi-module KMP projects.

## New Features

- **spmDevBuildAll**: Builds all XCFrameworks locally and generates
  Package.swift with local paths for development
- **kmmBridgePublishAll**: Publishes all modules and generates
  Package.swift with URLs for distribution
- **generatePackageSwift**: Generates Package.swift from published metadata

## Changes

- Add `KmmBridgeSpmPlugin` for root-level SPM management
- Add `KmmBridgeSpmExtension` for configuration options
- Add `SpmModuleMetadata` for JSON metadata exchange between modules
- Add `writeSpmMetadata` task to each module for metadata generation
- Disable module-level `spmDevBuild` when root SPM plugin is applied
- Add comprehensive documentation in docs/SPM_MULTI_MODULE.md

## Benefits

- No manual Package.swift editing required
- No need for `useCustomPackageFile` or `perModuleVariablesBlock` flags
- Unified workflow for both local development and CI publishing
- Automatic platform version resolution (takes maximum)
- Automatic Swift tools version resolution

faogustavo commented on kmmbridge/src/main/kotlin/co/touchlab/kmmbridge/dependencymanager/SpmDependencyManager.kt:
```diff
@@ -206,6 +247,10 @@ internal class SpmDependencyManager(
         if (useCustomPackageFile) return // No local dev when using a custom package file
 
+        // Skip if root SPM plugin is applied (use spmDevBuildAll instead)
+        val rootHasSpmPlugin = project.rootProject.extensions.findByName("kmmBridgeSpm") != null
```
Nit: Use the constant to prevent breaking if the name changes

```suggestion
        val rootHasSpmPlugin = project.rootProject.extensions.findByName(KmmBridgeSpmPlugin.EXTENSION_NAME) != null
```

faogustavo commented on kmmbridge/src/main/kotlin/co/touchlab/kmmbridge/spm/KmmBridgeSpmPlugin.kt:
```diff
@@ -0,0 +1,431 @@
+
+            // Depend on all module kmmBridgePublish tasks
+            kmmBridgeModules.forEach { module ->
+                val publishTask = module.tasks.findByName("kmmBridgePublish")
```
This is out of the scope, as it's already implemented this way, but if possible, try to create an object (or something similar) to store the task names. `uploadXCFramework`, `kmmBridgePublish`, `writeSpmMetadata`, etc).

Not a blocker :)

faogustavo reviewed (approved):
LGTM. Things seem to be working locally. I'll clean up my tests and push a branch with this new implementation to github.com/touchlab/KMMBridgeSPMQuickStart

faogustavo commented:
Adding copilot just for a second pair of eyes 😅

faogustavo committed:
Refactor SPM Plugin for more code reuse and addressing other code review comments

faogustavo committed:
Refactor SPM Plugin for more code reuse and addressing other code review comments

faogustavo commented:
Hey @hanrw 👋 

Can you check whether you enabled contributor permissions to update the PR? Or cherry-pick [this commit](https://github.com/touchlab/KMMBridge/commit/b49a1043d1c6ed1d5e8d40d5b6a8303b7743b7dd.patch) into your branch to fix the conversations in this PR?

hanrw committed:
Merge branch 'touchlab:feature/spm-multi-module-auto-generation' into feature/spm-multi-module-auto-generation

hanrw commented:
> Hey @hanrw 👋
> 
> Can you check whether you enabled contributor permissions to update the PR? Or cherry-pick [this commit](https://github.com/touchlab/KMMBridge/commit/b49a1043d1c6ed1d5e8d40d5b6a8303b7743b7dd.patch) into your branch to fix the conversations in this PR?

I’ve just pushed all the changes from the commit you mentioned

faogustavo commented:
@hanrw, can you update the brand again and double-check the Copilot comments, please? I think I addressed all of them, but a second pair of eyes would be helpful :)

hanrw committed:
refactor(spm): extract Package.swift generation to PackageSwiftGenerator
- Address Copilot review feedback for SPM multi-module support
- Move Package.swift content generation logic to a new class for testability
- Simplify KmmBridgeSpmPlugin by delegating package generation
- Enhance platform version parsing in SpmDependencyManager with max-version logic
- Remove obsolete local Package.swift generation code, use centralized generator class

hanrw commented:
> @hanrw, can you update the brand again and double-check the Copilot comments, please? I think I addressed all of them, but a second pair of eyes would be helpful :)

I've just pushed the latest changes based on the comments.
